modules/core/shared/src/main/scala/lucuma/core/enum/GmosDetector.scala

```diff
@@ -16,22 +23,60 @@ sealed abstract class GmosDetector(
   val tag: String,
   val shortName: String,
   val longName: String,
-  val northPixelSize: Angle,
-  val southPixelSize: Angle,
-  val shuffleOffset: Int,
-  val xSize: Int,
-  val ySize: Int,
-  val maxRois: Int
-) extends Product with Serializable
+  val pixelSize: Angle,
+  val shuffleOffset: Quantity[PosInt, Pixels],
+  val xSize: Quantity[PosInt, Pixels],
+  val ySize: Quantity[PosInt, Pixels],
+  val maxRois: PosInt
+) extends Product with Serializable {
+}
 
 object GmosDetector {
 
-  /** @group Constructors */ case object E2V extends GmosDetector("E2V", "E2V", "E2V", Angle.fromDoubleArcseconds(0.0727), Angle.fromDoubleArcseconds(0.0730), 1536, 6144, 4608, 4)
-  /** @group Constructors */ case object HAMAMATSU extends GmosDetector("HAMAMATSU", "Hamamatsu", "Hamamatsu", Angle.fromDoubleArcseconds(0.0809), Angle.fromDoubleArcseconds(0.0809), 1392, 6144, 4224, 5)
+  /** @group Constructors */ case object GmosNE2V extends GmosDetector(
+                               "GmosNE2V",
+                               "E2V",
+                               "E2V",
+                               Angle.fromMicroarcseconds(72700),
+                               1536.withRefinedUnit[Positive, Pixels],
+                               6144.withRefinedUnit[Positive, Pixels],
+                               4608.withRefinedUnit[Positive, Pixels],
+                               4
+                             )
+  /** @group Constructors */ case object GmosSE2V extends GmosDetector(
+                               "GmosSE2V",
+                               "E2V",
+                               "E2V",
+                               Angle.fromMicroarcseconds(73000),
+                               1536.withRefinedUnit[Positive, Pixels],
+                               6144.withRefinedUnit[Positive, Pixels],
+                               4608.withRefinedUnit[Positive, Pixels],
+                               4
+                             )
+  /** @group Constructors */ case object GmosNHammamtsu extends GmosDetector(
+                               "GmosNHAMAMATSU",
```

Comment:
I think this will turn into "GMOSNHAMAMATSU" in the GraphQL enum.  GMOS is a given maybe we should make it [NS]HAMAMATSU and [NS]E2V?  I don't know.